### PR TITLE
Adding unit testing

### PR DESCRIPTION
**Pull request by Lubandeo**
## Description

Added the unit tests to the project.

## Story Type

- New feature:white_check_mark:
- Change request
- Bug fix
- Tech

## How Can This Be Tested?

By runnning the test one by one or altogether and checking to see if all of them pass. 
## Evidences

Coverage:
image
image

## Checklist

- All unit Test passed  :white_check_mark:

### CODE REVIEW

**Review by jnicolaslaurenti — APPROVED** [review]

**Review by Georgina-Fernandez — APPROVED** [review]

File: app/src/main/java/com/mobile/tandil/javabaseproject/mvp/contract/ReservationContract.java [hunk 1]
```diff
@@ -44,8 +39,11 @@ interface ReservationView {
 
         void showEverythingOkMessage();
 
+        void showReservationSuccessful();
+
         void showExceptionInputStringMessage();
 
+
```
Comment (DiegoDiazGlobant) [inline app/src/main/java/com/mobile/tandil/javabaseproject/mvp/contract/ReservationContract.java]: Remove extra lines please

File: app/src/test/java/com/mobile/tandil/javabaseproject/ReservationPickersPresenterTest.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    public void initialize() {
+        MockitoAnnotations.initMocks(this);
+        presenter = new ReservationPickersPresenter(view, true);
+    }
+
+    @Test
+    public void testOnButtonOkPickersPressedCheckIn() {
+        presenter.onButtonOkPickersPressed(listenerReservation);
+        Mockito.verify(view).showCheckInReservation(listenerReservation);
+
```
Comment (DiegoDiazGlobant) [inline app/src/test/java/com/mobile/tandil/javabaseproject/ReservationPickersPresenterTest.java]: same here

File: app/src/test/java/com/mobile/tandil/javabaseproject/ReservationPickersPresenterTest.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        Mockito.verify(view).showCheckOutReservation(listenerReservation);
+
+    }
+
+    @Test
+    public void testOnButtonCancelPickersPressed() {
+
+        presenter.onButtonCancelPickersPressed();
+        Mockito.verify(view).dismissFragment();
+
```
Comment (DiegoDiazGlobant) [inline app/src/test/java/com/mobile/tandil/javabaseproject/ReservationPickersPresenterTest.java]: same here

File: app/src/test/java/com/mobile/tandil/javabaseproject/DialogFragmentPresenterTest.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    private static final String ZERO_PARKING = "0";
+
+    @Mock
+    private DialogFragmentContract.View view;
+
+    @Mock
+    private ListenerDialogFragment dialogFragmentListener;
+
+    @Before
+    public void initialize() {
```
Comment (SartiMau) [inline app/src/test/java/com/mobile/tandil/javabaseproject/DialogFragmentPresenterTest.java:30]: the mocks are never initialized. Fix this, please

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/mobile/tandil/javabaseproject/mvp/contract/ReservationContract.java (hunk 1 above)
Comment (Lubandeo) [reply app/src/main/java/com/mobile/tandil/javabaseproject/mvp/contract/ReservationContract.java]: Done

File: app/src/test/java/com/mobile/tandil/javabaseproject/ReservationPickersPresenterTest.java (hunk 2 above)
Comment (Lubandeo) [reply app/src/test/java/com/mobile/tandil/javabaseproject/ReservationPickersPresenterTest.java]: Done

File: app/src/test/java/com/mobile/tandil/javabaseproject/ReservationPickersPresenterTest.java (hunk 3 above)
Comment (Lubandeo) [reply app/src/test/java/com/mobile/tandil/javabaseproject/ReservationPickersPresenterTest.java]: Done

File: app/src/test/java/com/mobile/tandil/javabaseproject/DialogFragmentPresenterTest.java (hunk 4 above)
Comment (Lubandeo) [reply app/src/test/java/com/mobile/tandil/javabaseproject/DialogFragmentPresenterTest.java:30]: Done

**Review by SartiMau — APPROVED** [review]

**Review by DiegoDiazGlobant — APPROVED** [review]

**Pull request merged**